packages/flutter_tools/test/integration.shard/migrate_utils_test.dart

```diff
@@ -0,0 +1,223 @@
+      expect(await utils.isGitIgnored('other_file.dart', projectRootPath), false);
+    });
+
+    testWithoutContext('isGitRepo', () async {
+      expect(projectRoot.existsSync(), true);
+      expect(projectRoot.childDirectory('.git').existsSync(), false);
+      expect(await utils.isGitRepo(projectRootPath), false);
+      await utils.gitInit(projectRootPath);
+      expect(projectRoot.childDirectory('.git').existsSync(), true);
+
+      expect(await utils.isGitRepo(projectRootPath), true);
+
+      expect(await utils.isGitRepo(projectRoot.parent.path), false);
+    });
+
+    testWithoutContext('hasUncommitedChanges false on clean repo', () async {
+      expect(projectRoot.existsSync(), true);
+      expect(projectRoot.childDirectory('.git').existsSync(), false);
+      await utils.gitInit(projectRootPath);
+      expect(projectRoot.childDirectory('.git').existsSync(), true);
+
+      projectRoot.childFile('.gitignore')
+        ..createSync()
+        ..writeAsStringSync('ignored_file.dart', flush: true);
+
+      await processUtils.run(<String>['git', 'add', '.'], workingDirectory: projectRootPath);
+      await processUtils.run(<String>['git', 'commit', '-m', 'Initial commit'], workingDirectory: projectRootPath);
+
+      expect(await utils.hasUncommitedChanges(projectRootPath), false);
+    });
+
+    testWithoutContext('hasUncommitedChanges true on dirty repo', () async {
+      expect(projectRoot.existsSync(), true);
+      expect(projectRoot.childDirectory('.git').existsSync(), false);
+      await utils.gitInit(projectRootPath);
+      expect(projectRoot.childDirectory('.git').existsSync(), true);
+
+      projectRoot.childFile('some_file.dart')
+        ..createSync()
+        ..writeAsStringSync('void main() {}', flush: true);
+
+      expect(await utils.hasUncommitedChanges(projectRootPath), true);
+    });
+
+    testWithoutContext('diffFiles', () async {
+      expect(projectRoot.existsSync(), true);
+      expect(projectRoot.childDirectory('.git').existsSync(), false);
+      await utils.gitInit(projectRootPath);
+      expect(projectRoot.childDirectory('.git').existsSync(), true);
+
+      final File file1 = projectRoot.childFile('some_file.dart')
+        ..createSync()
+        ..writeAsStringSync('void main() {}\n', flush: true);
+
+      final File file2 = projectRoot.childFile('some_other_file.dart');
+
+      DiffResult result = await utils.diffFiles(file1, file2);
+      expect(result.diff, null);
+      expect(result.diffType, DiffType.deletion);
+      expect(result.exitCode, null);
+
+      result = await utils.diffFiles(file2, file1);
+      expect(result.diff, null);
+      expect(result.diffType, DiffType.addition);
+      expect(result.exitCode, null);
+
+      file2.createSync();
+      file2.writeAsStringSync('void main() {}\n', flush: true);
+
+      result = await utils.diffFiles(file1, file2);
+      expect(result.diff, '');
+      expect(result.diffType, DiffType.modification);
```

Comment:
Can this either be a new `DiffType.none` or even `result == null`? It seems counter-intuitive that the two files are the same but it the difftype is modification.

Reply:
I'll rename `modification` to something else instead. This is meant to represent the output of the `git diff` command, and if it runs and returns no differences, I would still want it to retain that information for the tool to work with.

Reply:
Renamed to `command` which better reflects that this state represents that we ran a diff command algo on the files, not whether there were any changes or not.